scripts/utilities/DEG_summary_plot.R

```diff
@@ -0,0 +1,44 @@
+# Find DEG tables
+
+files <- fs::dir_ls("./analysis/DEG_lists/", regexp = "DEG_summary.txt",
+                    recurse = T)
+files <- as.vector(files)
+
+# Exclude the top-level summary...
+files <- files[grep("DEG_lists/DEG_summary", files, invert = T)]
+
+# Function to read in one  at a time and wrangle data
+clean_data <- function(x) {
+  df <- read.table(x, sep = "\t", header = F, fill = T)  
+  df <- as.data.frame(t(df))
+  df[1,1] <- "Condition"
+  colnames(df) <- df[1,]
+  df <- df[2:nrow(df),]
+  # subset - shouldn't be needed but it is?
+  df <- df[,1:2]
+  final <-  reshape2::melt(df, id.vars=1)
+  colnames(final) <- c("Condition","Sample","DEGs")
+  final$DEGs <- as.numeric(final$DEGs)
+  final$Condition <- factor(final$Condition, levels = final$Condition)
+  return(final)
+}
+
+degs <- files %>%
+  map_dfr(clean_data)
+
+###################################################
+# Optional... Data cleaning examples
+# Some fun regex if you want to split things up...
+# This takes the last space in the string as the delimiter
+degs <- degs %>% tidyr::separate(Sample, c("Chemical","Timepoint"),
+                                 sep = "\\s+(?=\\S*$)" )
+#
+###################################################
+
+###################################################
+# Plot
+ggplot(degs, aes(x = Condition, y = DEGs, fill = Timepoint)) +
+  geom_bar(stat="identity", position="dodge") +
+  facet_wrap(~Chemical, scales = "free_x") +
+  theme(axis.text.x = element_blank()) # element_text(angle = 90))
+
```

Comment:
This isn't currently called from any of the workbooks right?

Reply:
Correct - has to be run manually. Just produces a figure showing DEGs from those awful and inconsistent DEG summary files. Works ok and probably takes a few minutes to run for a given project. I tested it on two different projects and was pleased with the result. Could definitely get more creative with it though.